src/main.ts

```diff
@@ -223,24 +225,32 @@ async function run(): Promise<void> {
     input.push(
       limit(async () => {
         const command = `app diff ${app.metadata.name} --revision=${github.context.payload.pull_request?.head?.sha}`;
-        try {
-          core.info(`Running: argocd ${command}`);
-          // ArgoCD app diff will exit 1 if there is a diff, so always catch,
-          // and then consider it a success if there's a diff in stdout
-          // https://github.com/argoproj/argo-cd/issues/3588
-          await argocd(command);
-        } catch (e) {
-          const res = e as ExecResult;
-          core.info(`stdout: ${res.stdout}`);
-          core.info(`stderr: ${res.stderr}`);
-          if (res.stdout) {
-            diffs.push({ app, diff: res.stdout });
-          } else {
-            diffs.push({
-              app,
-              diff: '',
-              error: e
-            });
+
+        for (let retry = 1; retry <= Number(RETRY_COUNT); retry++) {
+          try {
+            core.info(`Running (${retry}/${RETRY_COUNT}): argocd ${command}`);
+            // ArgoCD app diff will exit 1 if there is a diff, so always catch,
+            // and then consider it a success if there's a diff in stdout
+            // https://github.com/argoproj/argo-cd/issues/3588
+            await argocd(command);
+            break;
+          } catch (e) {
+            const res = e as ExecResult;
+            core.info(`stdout (${app.metadata.name}): ${res.stdout}`);
+            core.info(`stderr (${app.metadata.name}): ${res.stderr}`);
+            if (res.stdout) {
+              diffs.push({ app, diff: res.stdout });
+              break;
+            } else {
+              await new Promise(f => setTimeout(f, Number(RETRY_DELAY)));
+              if (retry === Number(RETRY_COUNT)) {
+                diffs.push({
+                  app,
+                  diff: '',
+                  error: e
+                });
+              }
```

Comment:
I guess you don't need to wait if `retry === Number(RETRY_COUNT`. I suggest switching these blocks
```suggestion
              if (retry === Number(RETRY_COUNT)) {
                diffs.push({
                  app,
                  diff: '',
                  error: e
                });
              }
              await new Promise(f => setTimeout(f, Number(RETRY_DELAY)));
```